main.go

```diff
@@ -38,77 +64,79 @@ func main() {
 		}
 
 		r.RequestURI = ""
-		req_a := *r
-		req_b := *r
 
-		ua, err := url.Parse(targets[0])
-		if err != nil {
-			panic(err)
-		}
-		ub, err := url.Parse(targets[1])
-		if err != nil {
-			panic(err)
-		}
+		// Create and send out an HTTP request to each target.
+		for i := range targets {
+			req := *r
+			req.URL = urls[i]
+			cancels[i] = make(chan struct{})
+			req.Cancel = cancels[i]
 
-		req_a.URL = ua
-		req_b.URL = ub
+			channel := channels[i]
 
-		cancel_a := make(chan struct{})
-		cancel_b := make(chan struct{})
+			go func() {
+				res, err := http.DefaultClient.Do(&req)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "request failed: %s\n", err)
+					return
+				}
 
-		req_a.Cancel = cancel_a
-		req_b.Cancel = cancel_b
-
-		go func() {
-			resp, err := http.DefaultClient.Do(&req_a)
-			if err != nil {
-				log.Printf("request to %s failed: %s", targets[0], err)
-				return
-			}
-
-			a <- resp
-		}()
-
-		go func() {
-			resp, err := http.DefaultClient.Do(&req_b)
-			if err != nil {
-				log.Printf("request to %s failed: %s", targets[1], err)
-				return
-			}
-
-			b <- resp
-		}()
+				channel <- res
+			}()
+		}
 
+		// Listen to all requests' cancel channels; issue a signal on a special failure channel.
 		failed := make(chan struct{})
 		go func() {
-			<-cancel_a
-			<-cancel_b
+			for i := range cancels {
```

Comment:
Good catch! I found some other issues as a result of investigating this. (see commits)